shared/reducers/fs.js

```diff
@@ -7,14 +7,31 @@ import * as Types from '../constants/types/fs'
 
 const initialState = Constants.makeState()
 
+const coalesceFolderUpdate = (
+  original: Types.FolderPathItem,
+  updated: Types.FolderPathItem
+): Types.FolderPathItem =>
+  // We don't want to override a loaded folder into pending, because otherwise
+  // next time user goes into that folder we'd show placeholders. We also don't
+  // want to simply use the original PathItem, since it's possible some
+  // metadata has updated. So use the new item, but reuse children and
+  // progress.
+  original.progress === 'loaded' && updated.progress === 'pending'
+    ? updated.withMutations(u => u.set('children', original.children).set('progress', 'loaded'))
```

Comment:
please put these `.set` calls on separate lines.

Reply:
Hmm no matter what I try, my linter keeps merging them back into single line. I guess linter doesn't do that when the line is too short?

Reply:
Your linter is too aggressive. Can you make it not rewrite the files?